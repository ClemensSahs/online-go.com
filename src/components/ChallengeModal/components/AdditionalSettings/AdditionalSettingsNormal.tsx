import * as React from "react";

import {
  _
} from "translate";
import * as data from "data";

import AdditionalSettingsBoardSize from './AdditionalSettingsBoardSize';
import AdditionalSettingsHeaderRanked from './AdditionalSettingsHeaderRanked';

import {
  AdditionalSettingsNormalProps,
  ChallengeModes
} from './types';

export const AdditionalSettingsNormal = ({
  enableCustomBoardSizes,
  mode,
  ranked,
  forkingGame,
  selectedBoardSize,
  gameHeight,
  gameWidth,
  challenge,
  onChangeRanked,
  onChangeAgaRanked,
  onChangeBoardSize,
  onChangeBoardWidth,
  onChangeBoardHeight,
  withAgaRanking
}: AdditionalSettingsNormalProps) => {
  if (forkingGame) {
    return null;
  }

  return (
    <div id="challenge-additional-settings" className="right-pane pane form-horizontal" role="form">
        <div>
            <AdditionalSettingsHeaderRanked
              ranked={challenge.game.ranked}
              isPrivate={challenge.game.private}
              label={_("Ranked")}
              onChange={onChangeRanked}
              id={'challenge-ranked'}
            />
            {withAgaRanking &&
              <AdditionalSettingsHeaderRanked
                ranked={challenge.aga_ranked}
                isPrivate={challenge.game.private}
                label={_("AGA Ranked")}
                onChange={onChangeAgaRanked}
                id={'challenge-aga-ranked'}
              />
            }
        </div>

        <AdditionalSettingsBoardSize
          enableCustomBoardSizes={enableCustomBoardSizes}
          selectedBoardSize={selectedBoardSize}
          gameWidth={gameHeight}
          gameHeight={gameWidth}
          onChangeBoardSize={onChangeBoardSize}
          onChangeBoardWidth={onChangeBoardWidth}
          onChangeBoardHeight={onChangeBoardHeight}
        />
    </div>
  );
};


export default AdditionalSettingsNormal;
