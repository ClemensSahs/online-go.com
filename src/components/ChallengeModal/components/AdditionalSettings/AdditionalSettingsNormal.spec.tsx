
import * as React from 'react';
import * as data from "data";

import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import AdditionalSettingsNormal from './AdditionalSettingsNormal';
import {
  ChallengeModes
} from '../../types';

jest.mock('data');
// jest.mock('data', () => ({
//     get: jest.fn((x: string, defaultValue) => {
//       console.log('mock '+ x);
//       if (x === 'config.aga_rankings_enabled') {
//         return true;
//       }
//       return defaultValue;
//     })
// }));

jest.mock('translate', () => ({
    _: jest.fn(x => x),
}));

const getBaseProps = () => {
  return {
    challenge: {
      game: {
        width: 19,
        height: 19
      }
    },
    conf: {
      selected_board_size: '19x19'
    },
    forking_game: '',
    mode: ChallengeModes.OPEN,
    ranked: false,
    update_demo_rules: (ev) => console.log(),
    update_board_size: (ev) => console.log(),
    update_board_width: (ev) => console.log(),
    update_board_height: (ev) => console.log(),
    update_ranked: (ev) => console.log(),
    update_aga_ranked: (ev) => console.log(),
  };
}

describe('ChallengeModes additional settings normal', () => {

  let mockedProps;
  beforeEach(() => {
    mockedProps = getBaseProps()
  });


  test('render additional settings', () => {

    render(<AdditionalSettingsNormal {...mockedProps} />);

    expect(screen.queryByLabelText('Ranked')).toBeTruthy();
    expect(screen.queryByLabelText('AGA Ranked')).toBeNull();
  });


  test('render additional settings with aga ranking', () => {
    (data.get as unknown as jest.Mock).mockImplementation((x: string, defaultValue?: any): any => {
      if (x === 'config.aga_rankings_enabled') {
        return true;
      }
      return defaultValue;
    });

    render(<AdditionalSettingsNormal {...mockedProps} />);

    expect(screen.queryByLabelText('Ranked')).toBeTruthy();
    expect(screen.queryByLabelText('AGA Ranked')).toBeTruthy();
  });

  test('render additional settings as folking game', () => {
    mockedProps.forking_game = true;

    render(<AdditionalSettingsNormal {...mockedProps} />);

    expect(screen.queryByLabelText('Ranked')).toBeNull();
    expect(screen.queryByLabelText('AGA Ranked')).toBeNull();
    expect(screen.queryByLabelText('Board Size')).toBeNull();
  });

  test('render additional settings with a custom board size', () => {
    mockedProps.conf.selected_board_size = "custom";

    render(<AdditionalSettingsNormal {...mockedProps} />);

    expect(screen.queryByTestId('challenge-goban-width')).toBeTruthy();
    expect(screen.queryByTestId('challenge-goban-height')).toBeTruthy();
  });
});
