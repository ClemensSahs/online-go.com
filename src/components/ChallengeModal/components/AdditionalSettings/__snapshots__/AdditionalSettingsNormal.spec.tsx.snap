// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componentSanityChecks for AdditionalSettingsNormal.basic matches the snapshot 1`] = `
<div
  className="right-pane pane form-horizontal"
  id="challenge-additional-settings"
  role="form"
>
  <div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="challenge-ranked"
      >
        Ranked
      </label>
      <div
        className="controls"
      >
        <div
          className="checkbox"
        >
          <input
            id="challenge-ranked"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group"
    id="challenge-board-size-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-board-size"
    >
      Board Size
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          id="challenge-board-size"
          onChange={[Function]}
          value="19x19"
        >
          <optgroup
            label="Normal Sizes"
          >
            <option
              value="19x19"
            >
              19x19
            </option>
            <option
              value="13x13"
            >
              13x13
            </option>
            <option
              value="9x9"
            >
              9x9
            </option>
          </optgroup>
          <optgroup
            label="Extreme Sizes"
          >
            <option
              disabled={true}
              value="25x25"
            >
              25x25
            </option>
            <option
              disabled={true}
              value="21x21"
            >
              21x21
            </option>
            <option
              disabled={true}
              value="5x5"
            >
              5x5
            </option>
          </optgroup>
          <optgroup
            label="Non-Square"
          >
            <option
              disabled={true}
              value="19x9"
            >
              19x9
            </option>
            <option
              disabled={true}
              value="5x13"
            >
              5x13
            </option>
          </optgroup>
          <optgroup
            label="Custom"
          >
            <option
              disabled={true}
              value="custom"
            >
              Custom Size
            </option>
          </optgroup>
        </select>
      </div>
    </div>
  </div>
</div>
`;

exports[`componentSanityChecks for AdditionalSettingsNormal.custom board size matches the snapshot 1`] = `
<div
  className="right-pane pane form-horizontal"
  id="challenge-additional-settings"
  role="form"
>
  <div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="challenge-ranked"
      >
        Ranked
      </label>
      <div
        className="controls"
      >
        <div
          className="checkbox"
        >
          <input
            id="challenge-ranked"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group"
    id="challenge-board-size-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-board-size"
    >
      Board Size
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          id="challenge-board-size"
          onChange={[Function]}
          value="custom"
        >
          <optgroup
            label="Normal Sizes"
          >
            <option
              value="19x19"
            >
              19x19
            </option>
            <option
              value="13x13"
            >
              13x13
            </option>
            <option
              value="9x9"
            >
              9x9
            </option>
          </optgroup>
          <optgroup
            label="Extreme Sizes"
          >
            <option
              disabled={false}
              value="25x25"
            >
              25x25
            </option>
            <option
              disabled={false}
              value="21x21"
            >
              21x21
            </option>
            <option
              disabled={false}
              value="5x5"
            >
              5x5
            </option>
          </optgroup>
          <optgroup
            label="Non-Square"
          >
            <option
              disabled={false}
              value="19x9"
            >
              19x9
            </option>
            <option
              disabled={false}
              value="5x13"
            >
              5x13
            </option>
          </optgroup>
          <optgroup
            label="Custom"
          >
            <option
              disabled={false}
              value="custom"
            >
              Custom Size
            </option>
          </optgroup>
        </select>
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-board-size-custom"
    />
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <input
          className="form-control"
          data-testid="challenge-goban-width"
          id="challenge-goban-width"
          max="25"
          min="1"
          onChange={[Function]}
          role="input"
          style={
            Object {
              "width": "3em",
            }
          }
          type="number"
          value={19}
        />
        <input
          className="form-control"
          data-testid="challenge-goban-height"
          id="challenge-goban-height"
          max="25"
          min="1"
          onChange={[Function]}
          role="input"
          style={
            Object {
              "width": "3em",
            }
          }
          type="number"
          value={19}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`componentSanityChecks for AdditionalSettingsNormal.with-aga-ranking matches the snapshot 1`] = `
<div
  className="right-pane pane form-horizontal"
  id="challenge-additional-settings"
  role="form"
>
  <div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="challenge-ranked"
      >
        Ranked
      </label>
      <div
        className="controls"
      >
        <div
          className="checkbox"
        >
          <input
            id="challenge-ranked"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group"
    id="challenge-board-size-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-board-size"
    >
      Board Size
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          id="challenge-board-size"
          onChange={[Function]}
          value="19x19"
        >
          <optgroup
            label="Normal Sizes"
          >
            <option
              value="19x19"
            >
              19x19
            </option>
            <option
              value="13x13"
            >
              13x13
            </option>
            <option
              value="9x9"
            >
              9x9
            </option>
          </optgroup>
          <optgroup
            label="Extreme Sizes"
          >
            <option
              disabled={true}
              value="25x25"
            >
              25x25
            </option>
            <option
              disabled={true}
              value="21x21"
            >
              21x21
            </option>
            <option
              disabled={true}
              value="5x5"
            >
              5x5
            </option>
          </optgroup>
          <optgroup
            label="Non-Square"
          >
            <option
              disabled={true}
              value="19x9"
            >
              19x9
            </option>
            <option
              disabled={true}
              value="5x13"
            >
              5x13
            </option>
          </optgroup>
          <optgroup
            label="Custom"
          >
            <option
              disabled={true}
              value="custom"
            >
              Custom Size
            </option>
          </optgroup>
        </select>
      </div>
    </div>
  </div>
</div>
`;

exports[`componentSanityChecks for AdditionalSettingsNormal.with-forking-game matches the snapshot 1`] = `null`;

exports[`componentSanityChecks for AdditionalSettingsNormal.without-aga-ranking matches the snapshot 1`] = `
<div
  className="right-pane pane form-horizontal"
  id="challenge-additional-settings"
  role="form"
>
  <div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="challenge-ranked"
      >
        Ranked
      </label>
      <div
        className="controls"
      >
        <div
          className="checkbox"
        >
          <input
            id="challenge-ranked"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="challenge-aga-ranked"
      >
        AGA Ranked
      </label>
      <div
        className="controls"
      >
        <div
          className="checkbox"
        >
          <input
            id="challenge-aga-ranked"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group"
    id="challenge-board-size-group"
  >
    <label
      className="control-label"
      htmlFor="challenge-board-size"
    >
      Board Size
    </label>
    <div
      className="controls"
    >
      <div
        className="checkbox"
      >
        <select
          className="challenge-dropdown form-control"
          id="challenge-board-size"
          onChange={[Function]}
          value="19x19"
        >
          <optgroup
            label="Normal Sizes"
          >
            <option
              value="19x19"
            >
              19x19
            </option>
            <option
              value="13x13"
            >
              13x13
            </option>
            <option
              value="9x9"
            >
              9x9
            </option>
          </optgroup>
          <optgroup
            label="Extreme Sizes"
          >
            <option
              disabled={true}
              value="25x25"
            >
              25x25
            </option>
            <option
              disabled={true}
              value="21x21"
            >
              21x21
            </option>
            <option
              disabled={true}
              value="5x5"
            >
              5x5
            </option>
          </optgroup>
          <optgroup
            label="Non-Square"
          >
            <option
              disabled={true}
              value="19x9"
            >
              19x9
            </option>
            <option
              disabled={true}
              value="5x13"
            >
              5x13
            </option>
          </optgroup>
          <optgroup
            label="Custom"
          >
            <option
              disabled={true}
              value="custom"
            >
              Custom Size
            </option>
          </optgroup>
        </select>
      </div>
    </div>
  </div>
</div>
`;
